Tools/2MGFX/MonoGame.Framework/GraphicsDevice.cs

```diff
@@ -1,7 +1,18 @@
-﻿namespace Microsoft.Xna.Framework.Graphics
+﻿using System;
+
+namespace Microsoft.Xna.Framework.Graphics
 {
     // Dummy class for 2MGFX to compile.
     public class GraphicsDevice
     {
+        internal void AddResourceReference(WeakReference resourceReference)
+        {
+            
+        }
+
+        internal void RemoveResourceReference(WeakReference resourceReference)
+        {
+            
+        }
```

Comment:
> I haven't dug into it to see why we need dummy classes

We need them because 2MGFX uses some features of `MonoGame.Framework`...  nothing platform specific... just some enums and the render state classes.

If I had referenced the `MonoGame.Framework` assembly at the time 2MGFX would require assemblies for Lidgren, Direct Input, and other junk it never ever needs.  By just sharing source code from the framework and adding a dummy class for `GraphicsDevice` I avoided that dependency.

Now Lidgren junk is separate from the main `MonoGame.Framework`, so things are better.  However there was discussion of breaking up `MonoGame.Framework` further, but I don't know if that will happen or is a good idea.

> Do people still use 2MGFX as a standalone executable?

It exists because that is how I developed the stock effects.  If you are loading an effect from the `Effect(byte[[)` constructor you need an effect not packaged in XNB form.  The sock effects are embedded resources in the 2MGFX format.